pkg/operator/etcdendpointscontroller/etcdendpointscontroller.go

```diff
@@ -83,9 +83,29 @@ func (c *EtcdEndpointsController) sync(ctx context.Context, syncCtx factory.Sync
 	return nil
 }
 
-func (c *EtcdEndpointsController) syncConfigMap(ctx context.Context, recorder events.Recorder) error {
+func (c *EtcdEndpointsController) syncConfigMap(recorder events.Recorder) error {
+	bootstrapComplete, err := isBootstrapComplete(c.configmapLister, c.operatorClient)
+	if err != nil {
+		return fmt.Errorf("couldn't determine bootstrap status: %w", err)
+	}
+
 	required := configMapAsset()
 
+	// If the bootstrap IP is present on the existing configmap, either copy it
+	// forward or remove it if possible so clients can forget about it.
+	if existing, err := c.configmapLister.ConfigMaps(operatorclient.TargetNamespace).Get("etcd-endpoints"); err == nil {
```

Comment:
do we care if the `errors.IsNotFound(err)`? looks like we check
```
err == nil
 !errors.IsNotFound(err)
```
But what if the cm does not exist shouldn't we report that error? Should this be a switch?

Reply:
The CM doesn't exist prior to 4.5. If it's not found, we must create it.

In a new 4.5+ cluster, the CM is created during bootstrap, so if the CM doesn't exist but bootstrapping is still in-progress, that could be construed as an error, but that doesn't seem likely at this point and we would want to smoke out that issue separately IMO

Reply:
Although since this patch is now targeting 4.6, I guess we could go back to the "cm must always exist" invariant. As I have it here we could feasibly backport.

Reply:
> In a new 4.5+ cluster, the CM is created during bootstrap, so if the CM doesn't exist but bootstrapping is still in-progress, that could be construed as an error, but that doesn't seem likely at this point and we would want to smoke out that issue separately IMO

OK lets track that, thanks